contracts/erc7683/ERC7683Depositor.sol

```diff
@@ -0,0 +1,293 @@
+    /// @dev Intended to improve standardized integration of various order types and settlement contracts
+    /// @param order The CrossChainOrder definition
+    /// @param fillerData Any filler-defined data required by the settler
+    /// @return ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+    function resolve(CrossChainOrder memory order, bytes memory fillerData)
+        external
+        view
+        returns (ResolvedCrossChainOrder memory);
+}
+
+/**
+ * @notice Permit2Depositor processes an external order type and translates it into an AcrossV3 deposit.
+ */
+contract ERC7683OrderDepositor is ISettlementContract {
+    using SafeERC20 for IERC20;
+
+    // Unique Across nonce
+    uint256 depositId;
+
+    // SpokePool that this contract can deposit to.
+    V3SpokePoolInterface public immutable SPOKE_POOL;
+
+    // Permit2 contract for this network
+    IPermit2 public immutable PERMIT2;
+
+    // quoteBeforeDeadline is subtracted from the deadline to get the quote timestamp.
+    // This is a somewhat arbitrary conversion, but order creators need some way to precompute the quote timestamp.
+    uint256 public immutable QUOTE_BEFORE_DEADLINE;
+
+    /**
+     * @notice Construct the Permit2Depositor.
+     * @param _spokePool SpokePool that this contract can deposit to.
+     * @param _permit2 Permit2 contract
+     * @param _quoteBeforeDeadline quoteBeforeDeadline is subtracted from the deadline to get the quote timestamp.
+     */
+    constructor(
+        V3SpokePoolInterface _spokePool,
+        IPermit2 _permit2,
+        uint256 _quoteBeforeDeadline
+    ) {
+        SPOKE_POOL = _spokePool;
+        PERMIT2 = _permit2;
+        QUOTE_BEFORE_DEADLINE = _quoteBeforeDeadline;
+    }
+
+    function initiate(
+        CrossChainOrder memory order,
+        bytes memory signature,
+        bytes memory fillerData
+    ) external {
+        // Ensure that order was intended to be settled by Across.
+        require(order.settlementContract == address(this));
+        require(order.originChainId == block.chainid);
+
+        // Extract Across-specific params.
+        (AcrossOrderData memory acrossOrderData, ResolvedCrossChainOrder memory resolvedOrder, ) = _resolve(
+            order,
+            fillerData
+        );
+
+        // Require that the order has a single input and output.
+        require(
+            resolvedOrder.swapperInputs.length == 1 &&
+                resolvedOrder.swapperOutputs.length == 1 &&
+                resolvedOrder.fillerOutputs.length == 1
+        );
+
+        // Verify Permit2 signature and pull user funds into this contract
+        _processPermit2Order(PERMIT2, order, resolvedOrder, signature);
+
+        IERC20(resolvedOrder.swapperInputs[0].token).safeIncreaseAllowance(
+            address(SPOKE_POOL),
+            resolvedOrder.swapperInputs[0].amount
+        );
+        SPOKE_POOL.depositV3(
```

Comment:
Thoughts on integrating directly into the SpokePool contract rather than making an external call?

Pros: cheaper on gas, people are permitting the spoke pool contract directly
Cons: more complexity in the spoke pool

Reply:
we're also going to run into bytecode limits